# Econometric 322 Lab \#1

Collaboration Policy

    1. Study groups are allowed but I expect students to understand and complete their own assignments and to hand in one assignment per student.
    2. If you worked in a group, please put the names of your study group on your assignment on top.
    3. Just like all other classes at Rutgers, the student Honor Code is taken seriously.
    
    The submitted assignmenst must be your work.

## <font color = blue> Assignment </font>

Collect data from the Economic Report of the President: 2004, Table B-96. You just need New Business Incorporations, Total Number of Failures, and Total Amount of Current Liabilities for 1955 to 1997. Import the data into the notebook.  Name the variables as you wish but whatever names you use, document them in the Data Dictionary.  See the Lab Manual and the class notes for examples of data dictionaries.

# <font color = red> Documentation </font>

## <font color = blue> Abstract </font>

*The goal of this lab will be to examine the relationships between liabilities and business failures between 1955 to 1997. Failures and liabilities were strongly positively correlated, which is logical, as failing firms lacking revenue would need debt just to cover their operating costs. On the other hand, the correlation could be purely coincedental, as both failures and debt may grow over time.*

## <font color = blue> Data Dictionary </font>

Variable|Values|Source|Mnemonic
-|-|-|-
New Business Incorporations|Incorporations|Economic Report of the President: 2004 Table B-96|nbi
Total Number of Failures|Failures|Economic Report of the President: 2004 Table B-96|tnf
Total Amount of Current Liabilities|Millions of 2004 Dollars|Economic Report of the President: 2004 Table B-96|tcl

# <font color = red> Pre-lab Questions </font>

Before you do any work, please think about the concepts: business incorporations, failures, and liabilities in the U.S. from 1955 to 1997. In particular, think how you would answer the following if called on in class:

## <font color = blue> What type of data is this and why (i.e., source and domain)? </font>

This is time-series data, because we're comparing a few metrics for the United States over time. This data is secondary data because it comes from observational studies done by the federal government.

## <font color = blue> What relationships would you expect to exist among these variables?  Explain your answer. </font>

I would assume New Business Incorporations to rise during expansions, and fall during recessions, as strong growth would make people more optimistic and more likely to start businesses. Conversely, Total Number of Failures would decline during expansions and rise during recessions, as a bad economy would mean fewer sales. Ergo, those two variables would seem negatively correlated, even though neither necessarily causes the other. I would expect Total Amount of Current Liabilities to rise during expansions, as growing firms are eager to go in debt to expand, and to fall during recessions when there are fewer firms, and those firms make fewer investments. Total Amount of Current Liabilities I would expect to appear negatively correlated with Total Number of Failures and positively correlated with New Business Incorporations.

## <font color = blue> Write a testable hypothesis for failures as a function of liabilities.  Explain your answer. </font>

b1 is defined as the correlation between Total Number of Failures and Total Amount of Current Liabilities.
*H0*: b1 =0
*Ha*: b1 < 0

When Total Amount of Current Liabilities increases, Total Number of Failures decreases, so b1 would be negative. This would be because an increase in Total Amount of Current Liabilities means an increase in the number of firms who use debt to finance their operations, or each firm is taking on more debt to finance new investments, indicating a strong economy. A strong economy would then mean strong spending, and firms would be making a lot of sales, so fewer firms would fail, so the Total Number of Failures would be low.

## <font color = blue> Write a linear model for failures as a function of liabilities.

(*Total Number of Failures*)=b0+(b1 x *Total Amount of Current Liabilities in millions of dollars* )+error                                                       
error ~ N(0,sigma^2). We assume the error is normally distributed.

## <font color = blue> How do you think these variables will change over time? Do you think they will grow or decline or stay constant?  Explain your answer. </font>

I expect all three of these variables, New Business Incorporations, Total Number of Failures, and Total Amount of Current Liabilities to fluctuate somewhat. However, in the long run, I expect them to grow. As the economy and population grow, the number of businesses that are started and then fail will necessarily increase as well, as even if the average American is equally likely to start a business over time, and a business is equally likely to fail over time, the greater quantity of Americans will mean an increase in new businesses, and an increase in the number of businesses will mean in increase in the number of failures. Similarly, even if the average business is equally likely to take on the same average quantity debt over time, an increase in the number of businesses would mean an increase in the amount of total liabilities.

## <font color = blue> Would the business cycle have any impact on them?  Can you state any hypotheses about the business cycle profle of these variables?  What is a "business cycle profle?"  Explain your answers. </font>

I theorize that during expansions, Total Number of Failures declines, as a strong economy boosts sales and profits, and businesses do better. On the converse, during recessions, Total Number of Failures will increase, as the weak economy will starve businesses of revenue and they will go under. During expansions, optimism will make entrepreneurs will be more likely to start businesses, and New Business Incorporations will increase, and it will correspondingly fall during recessions. During periods of growth, businesses will also be more likely to go into debt to finance new investments, because they will be more confident about their investments paying off, so Total Amount of Current Liabilities will increase. During periods of contraction, businesses will be less likely to go into debt, and Total Amount of Current Liabilities will decrease. The business cycle profile of a variable is how it reacts to the various stages of the business cycle an economy may be in.

## <font color = blue> Can you think of any other variables you could create using just these three variables that may be interesting and meaningful? Describe them and state why they would be interesting and meaningful. </font>

I would subtract Total Number of Failures from New Business Incorporations to yield Net Increase in Businesses. The value of this would be to reflect the total increase in number of businesses, and would be a better metric than New Business Incorporations for gauging economic activity by business quantity. A period of severe economic stress may have a small uptick of business formation due to other factors, such as population growth, or changes in fiscal or regulatory policy. The increase in business Total Number of Failures would presumably be much greater. Net Increase in Businesses would be negative, but New Business Incorporations would be positive, meaning Net Increase in Businesses might be a better indicator.

# <font color = red> Tasks and Questions </font>

## <font color = blue> Load the Pandas package and give it an alias. </font>

In [4]:
##
## Enter the code here
##
import pandas as pd

## <font color = blue> Import the data. </font>

In [5]:
##
## Enter the code here
##
df=pd.read_csv(r"erp.csv") #the data was downloaded cleaned and saved as erp.csv

## <font color = blue> Print your data </font>

### <font color = blue> Print the first five (5) records. </font>

In [6]:
##
## Enter the code here
##

df.head()

year  netindex       nbi   bfr      tnf  underhundred  overhundred  \
0  1955.0      96.6  139915.0  42.0  10969.0       10113.0        856.0   
1  1956.0      94.6  141163.0  48.0  12686.0       11615.0       1071.0   
2  1957.0      90.3  137112.0  52.0  13739.0       12547.0       1192.0   
3  1958.0      90.2  150781.0  56.0  14964.0       13499.0       1465.0   
4  1959.0      97.9  193067.0  52.0  14053.0       12707.0       1346.0   

     tcl  underhundredliab  overhundredliab  
0  449.4             206.4            243.0  
1  562.7             239.8            322.9  
2  615.3             267.1            348.2  
3  728.3             297.6            430.7  
4  692.8             278.9            413.9

### <font color = blue> Print the last five (5) records.

In [7]:
##
## Enter the code here
##
df.tail()

year  netindex       nbi    bfr      tnf  underhundred  overhundred  \
38  1993.0     121.1  706537.0  109.0  86133.0       61188.0      24945.0   
39  1994.0     125.5  741778.0   86.0  71558.0       50814.0      20744.0   
40  1995.0       0.0  766988.0   82.0  71128.0       49495.0      21633.0   
41  1996.0       0.0  786482.0   80.0  71931.0       49667.0      22264.0   
42  1997.0       0.0  798779.0   88.0  83384.0       56050.0      27334.0   

        tcl  underhundredliab  overhundredliab  
38  47755.5             947.6          46807.9  
39  28977.9             845.0          28132.9  
40  37283.6             866.1          36417.4  
41  29568.7             914.9          28653.8  
42  37436.9            1111.3          36325.6

## <font color = blue> Reset the row index to have year as the index and print the first five (5) records. </font>

In [8]:
##
## Enter the code here
##

del df['overhundredliab']
del df['underhundredliab']
del df['overhundred']
del df['underhundred']
del df['netindex']
del df['bfr']

#irrelevant data series were deleted in order to make things more readable

df.set_index('year').head()

nbi      tnf    tcl
year                            
1955.0  139915.0  10969.0  449.4
1956.0  141163.0  12686.0  562.7
1957.0  137112.0  13739.0  615.3
1958.0  150781.0  14964.0  728.3
1959.0  193067.0  14053.0  692.8

## <font color = blue> Create and print a new variable that is the ratio of failures to liabilities. </font>

In [9]:
##
## Enter the code here
##
# ratio of failures to liabilities will have the name rftl
df['rftl']=df['tnf']/df['tcl']
df['rftl']

0     24.408100
1     22.544873
2     22.328945
3     20.546478
4     20.284353
5     16.455359
6     15.663701
7     13.004285
8     10.626941
9     10.157237
10    10.224711
11     9.425561
12     9.772368
13    10.240170
14     8.015060
15     5.693400
16     5.386822
17     4.782522
18     4.065518
19     3.247519
20     2.609926
21     3.197290
22     2.558395
23     2.492093
24     2.835720
25     2.533279
26     2.414596
27     1.595562
28     1.949493
29     1.779313
30     1.550001
31     1.377694
32     1.759917
33     1.442827
34     1.189757
35     1.082254
36     0.910299
37     1.029173
38     1.803625
39     2.469399
40     1.907756
41     2.432674
42     2.227321
Name: rftl, dtype: float64

## <font color = blue> Interpret the ratio you just created. </font>

This is a metric of failures to liabilities. When it increases, that means firms are failing at increasing frequency or taking on debt at lower levels. I predict that during expansions, lending increases, and failures decrease. Ergo, a high ratio of failures to liabilities indicates a recession and a low ratio indicates an expansion.

## <font color = blue> Create and print a new variable that is incorporations lagged one period. </font>

In [10]:
##
## Enter the code here
##
#new variable will be called nbiLag
df['nbiLag']=df['nbi']
df['nbiLag'].shift()

0          NaN
1     139915.0
2     141163.0
3     137112.0
4     150781.0
5     193067.0
6     182713.0
7     181535.0
8     182057.0
9     186404.0
10    197724.0
11    203897.0
12    200010.0
13    206569.0
14    233635.0
15    274267.0
16    264209.0
17    287577.0
18    316601.0
19    329358.0
20    319149.0
21    326345.0
22    375766.0
23    436170.0
24    478019.0
25    524565.0
26    533520.0
27    581242.0
28    566942.0
29    600420.0
30    634991.0
31    664235.0
32    702738.0
33    685572.0
34    685095.0
35    676565.0
36    647366.0
37    628604.0
38    666800.0
39    706537.0
40    741778.0
41    766988.0
42    786482.0
Name: nbiLag, dtype: float64

## <font color = blue> Create and print a new variable that is year-over-year percent change in incorporations. </font>

In [11]:
##
## Enter the code here
##
# this new variable will be named nbiChng
df['nbiChng']=df['nbi'].pct_change()
df['nbiChng']

0          NaN
1     0.008920
2    -0.028697
3     0.099692
4     0.280446
5    -0.053629
6    -0.006447
7     0.002875
8     0.023877
9     0.060728
10    0.031220
11   -0.019064
12    0.032793
13    0.131026
14    0.173912
15   -0.036672
16    0.088445
17    0.100926
18    0.040294
19   -0.030997
20    0.022547
21    0.151438
22    0.160749
23    0.095947
24    0.097373
25    0.017071
26    0.089447
27   -0.024602
28    0.059050
29    0.057578
30    0.046054
31    0.057966
32   -0.024427
33   -0.000696
34   -0.012451
35   -0.043158
36   -0.028982
37    0.060763
38    0.059594
39    0.049878
40    0.033986
41    0.025416
42    0.015635
Name: nbiChng, dtype: float64

Math function summary:

exp, *, /, +, -, df.shift(), df.pct_change()

## <font color = blue> Print a correlation matrix </font>

In [12]:
##
## Enter the code here
##
df.corr()
import seaborn as sns
print(sns)

year       nbi       tnf       tcl      rftl    nbiLag   nbiChng
year     1.000000  0.975994  0.811147  0.749606 -0.866306  0.975994 -0.143281
nbi      0.975994  1.000000  0.816394  0.733417 -0.797055  1.000000 -0.124952
tnf      0.811147  0.816394  1.000000  0.916073 -0.489270  0.816394 -0.248633
tcl      0.749606  0.733417  0.916073  1.000000 -0.519645  0.733417 -0.250257
rftl    -0.866306 -0.797055 -0.489270 -0.519645  1.000000 -0.797055  0.095814
nbiLag   0.975994  1.000000  0.816394  0.733417 -0.797055  1.000000 -0.124952
nbiChng -0.143281 -0.124952 -0.248633 -0.250257  0.095814 -0.124952  1.000000

# <font color = red> Post-Lab Questions </font>

## <font color = blue> What can you observe about the correlation matrix? Explain. </font>

This correlation matrix only contains values from -1 to 1. There is a diagonal series from the top left to the bottom right corner of the matrix where the correlation is consistently 1.0. This is because the correlation between a variable and itself is always 1. Every value below the diagonal is also reflected above the diagonal, because the correlation between X and Y is the same as the correlation between Y and X. The strongest positive correlation was the correlation between year and new business incorporations (R= 0.975994). The strongest negative correlation was between year and ratio of failures to liabilities (R=-0.866306). The weakest positive correlation was between ratio of failures to liabilities and year-over-year change in liabilities (R=0.095814). The weakest negative correlation was between year-over-year percent change in liabilities and new business incorporations (R=-0.124952).

## <font color = blue> What is the relationship between failures and incorporations? Explain. </font>

New Business Incorporations and Total Number of Failures are positively correlated (R=.816394). This does not imply causation either way. I theorize that as the population increases over time, more people start businesses, so there will be more businesses and failures of such. Thus, both variables would increase over time. This is one possible reason failures and incorporations could be correlated, but not causally related. On the other hand, Total Number of Failures and year-over-year percent change in incorporations are weakly negatively correlated (R=-0.248633). 

## <font color = blue> Do you see anything to support your testable hypothesis? Explain. </font>

My testable hypothesis was that the correlation coefficient between Total Number of Failures and Total Amount of Current Liabilities would be negative. In fact, the correlation between those two variables is strongly positive (R=0.916073). We fail to reject the null hypothesis and my testable hypothesis has been falsified. 

## <font color = blue> Does the amount of liabilities have anything to do with incorporations and failures? How and why? </font>

Total Amount of Current Liabilities and New Business Incorporations are positively correlated (R=0.733417). Total Amount of Current Liabilities and Total Number of Failures are also positively correlated (R=0.916073). Neither of the correlations prove causation. I theorize that over time, population increases, and thus there are more businesses opening. At the same time, I predict that over time, the real money supply also increases, and thus more loans can be made, and thus Total Amount of Current Liabilities will increase. This is only one possible explanation why these variables could appear correlated. On the contrary, there could be causal relationship. New businesses may be eager to accumulate debt for their early growth, and thus an increase in New Business Incorporations would mean an increase in liabilities. An increase in liabilities may also reflect that firms revenues are insufficient to cover their costs, and these failing firms will eventually go bankrupt due to their high debt loads, leading to an increase in Total Number of Failures. The data does not verify or falsify these theories, nor are any of them mutually exclusive.